i18n-helpers/src/lib.rs

```diff
@@ -219,23 +239,53 @@ pub fn group_events<'a>(events: &'a [(usize, Event<'a>)]) -> Vec<Group<'a>> {
                 State::Translate(start) => {
                     if ctx.skip_next_group {
                         (
-                            Group::Skip(&events[start..idx]),
+                            Group::Skip(
+                                &events[start..idx],
+                                GroupExtra {
+                                    translator_comments: vec![],
+                                },
+                            ),
                             ctx.clear_skip_next_group(),
                         )
                     } else if is_nontranslatable_codeblock_group(&events[start..idx]) {
-                        (Group::Skip(&events[start..idx]), ctx)
+                        (
+                            Group::Skip(
+                                &events[start..idx],
+                                GroupExtra {
+                                    translator_comments: vec![],
+                                },
+                            ),
+                            ctx,
+                        )
                     } else {
-                        (Group::Translate(&events[start..idx]), ctx)
+                        (
+                            Group::Translate(
+                                &events[start..idx],
+                                GroupExtra {
+                                    translator_comments: ctx.translator_comments.clone(),
+                                },
+                            ),
+                            ctx.clear_translator_comments(),
```

Comment:
There is an interesting pattern here: you clone `ctx.translator_comments` and then you clear them right after. So you could skip the clone if you moved the comments instead.

You can do this with a `std::mem::swap`: it can move values between two places without deallocating either.

Reply:
Acknowledged; replaced with use of std::mem::take().